dipy/align/imaffine.py

```diff
@@ -0,0 +1,777 @@
+                                        static_spacing, self.ss_sigma_factor,
+                                        False)
+
+    def optimize(self, static, moving, transform, x0, static_grid2space=None,
+                 moving_grid2space=None, prealign=None):
+        r''' Starts the optimization process
+
+        Parameters
+        ----------
+        static: array, shape (S, R, C) or (R, C)
+            the image to be used as reference during optimization.
+        moving: array, shape (S, R, C) or (R, C)
+            the image to be used as "moving" during optimization. It is
+            necessary to pre-align the moving image to ensure its domain
+            lies inside the domain of the deformation fields. This is assumed
+            to be accomplished by "pre-aligning" the moving image towards the
+            static using an affine transformation given by the 'prealign'
+            matrix
+        transform: instance of Transform
+            the transformation with respect to whose parameters the gradient
+            must be computed
+        x0: array, shape (n,)
+            parameters from which to start the optimization. If None, the
+            optimization will start at the identity transform. n is the
+            number of parameters of the specified transformation.
+        static_grid2space: array, shape (dim+1, dim+1), optional
+            the voxel-to-space transformation associated with the static
+            image. The default is None, implying the transform is the
+            identity.
+        moving_grid2space: array, shape (dim+1, dim+1), optional
+            the voxel-to-space transformation associated with the moving
+            image. The default is None, implying the transform is the
+            identity.
+        prealign: string, or matrix, or None, optional
+            If string:
+                'mass': align centers of gravity
+                'origins': align physical coordinates of voxel (0,0,0)
+                'centers': align physical coordinates of central voxels
+            If matrix:
+                array, shape (dim+1, dim+1).
+            If None:
+                Start from identity.
+            The default is None.
+
+        Returns
+        -------
+        T : array, shape(dim+1, dim+1)
+            the matrix representing the optimized affine transform
+        '''
+        self._init_optimizer(static, moving, transform, x0, static_grid2space,
+                             moving_grid2space, prealign)
+        del prealign  # Now we must refer to self.prealign
+
+        # Multi-resolution iterations
+        original_static_grid2space = self.static_ss.get_affine(0)
+        original_moving_grid2space = self.moving_ss.get_affine(0)
+
+        for level in range(self.levels - 1, -1, -1):
+            self.current_level = level
+            max_iter = self.level_iters[level]
+            print('Optimizing level %d [max iter: %d]' % (level, max_iter))
```

Comment:
> Make print optional with something like verbose flag?

Sure!, I will the same align.VerbosityLevels as in imwarp. Right now, it only prints the current level, but in the future we may add some debug information.